### PR TITLE
[flutter_tools] force page refresh when hot restarting in profile/release mode

### PR DESCRIPTION
**Pull request by jonahwilliams**
If we're running in profile or release mode, force the page to do a full refresh so it does not load the already cached JavaScript

### CODE REVIEW

File: packages/flutter_tools/lib/src/build_runner/resident_web_runner.dart [hunk 1]
```diff
@@ -808,10 +812,15 @@ class _DwdsResidentWebRunner extends ResidentWebRunner {
...
-        await wipConnection.sendCommand('Page.reload');
+        // On non-debug builds, a hard refresh is required to ensure the
+        // up to date sources are loaded.
+        await wipConnection?.sendCommand('Page.reload', <String, Object>{
+          'ignoreCache': !debuggingOptions.buildInfo.isDebug,
+        });
         status.stop();
         return OperationResult.ok;
       } catch (err) {
+        print('ERRR: $err');
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/build_runner/resident_web_runner.dart]: nit: typo in ERRR

**Review by yjbanov — APPROVED** [review]

File: packages/flutter_tools/lib/src/build_runner/resident_web_runner.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_runner/resident_web_runner.dart]: Oops, fixed

File: packages/flutter_tools/test/general.shard/resident_web_runner_cold_test.dart [hunk 2]
```diff
@@ -126,6 +129,38 @@ void main() {
     expect(result.message, contains('Failed to recompile application.'));
   }));
 
+  test('Correctly peforms a full refresh on attached chrome device.', () => testbed.run(() async {
```
Comment (ferhatb) [inline packages/flutter_tools/test/general.shard/resident_web_runner_cold_test.dart]: typo :  peforms -> performs

**Review by ferhatb — APPROVED** [review]

File: packages/flutter_tools/test/general.shard/resident_web_runner_cold_test.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/resident_web_runner_cold_test.dart]: Done

**Pull request merged**